web/src/app/apply-token/apply-token.component.ts

```diff
@@ -0,0 +1,28 @@
+import {Component, OnInit} from '@angular/core';
+import {ActivatedRoute, Router} from '@angular/router';
+import {HttpParams} from '@angular/common/http';
+import {TokenService} from '../_services/token.service';
+
+@Component({
+  selector: 'imprint-app-apply-token',
+  templateUrl: './apply-token.component.html',
+  styleUrls: ['./apply-token.component.scss']
+})
+export class ApplyTokenComponent implements OnInit {
+
+  constructor(
+    private route: ActivatedRoute,
+    private router: Router,
+    private tokenService: TokenService
+  ) {
+  }
+
+  ngOnInit() {
+    this.route.fragment.subscribe((fragment: string) => {
```

Comment:
Maybe to use `AcitvatedRouteSnapshot` instead of subscribing for a single value?

Reply:
Agree. It was port but need to be changed.)